### ISSUE TITLE
How does inputEnabledProperty apply to dropperNode?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/ph-scale#215]
Related to https://github.com/phetsims/ph-scale/issues/214... 

I see that inputEnabledProperty was not added to element `dropperNode`, and @kathy-phet had mentioned that as a specific use case.  The specific element is `phScale.macroScreen.view.dropperNode`. 

If I instrument `dropperNode.inputEnabledProperty`, that will disable input for EVERYTHING related to the dropper, including the button.   The drag listener is added to the entire dropper, so that you can drag anywhere on the dropper, including on the button.  There is no subcomponent that deals exclusively with dragging, and therefore no way to disable only dragging.  I'm not sure what to do, if anything.  Please advise.

If we decide to instrument `dropperNode.inputEnabledProperty`, should setting that to false hide the button on the dropper?

What we do here should apply to beers-law-lab.

**Comment by kathy-phet** [comment]
I'm not exactly sure who to tag on this issue. This seems really similar to the laser in bending light, and while I could punt here for this sim, in Bending Light it is essential to be able to use inputEnabledProperty in this way and still be able to turn on/off the laser. 

@jonathanolson @zepumph - Do either of you have a recommendation for Chris here in terms of how to proceed to get the desired behavior of not allowing dragging on the dropper node, but allowing the button to still work?

**Comment by pixelzoom** [comment]
To expand on the technical details and requirements here...

BLLDropperNode extends EyeDropperNode.  EyeDropperNode includes the "body" Node and the momentary button, and there's currently no way to access just the body via its API.   So BLLDropperNode adds a DragListener via `this.addInputListener`.  I could resolve the issue by making the body part of the `@protected` API, then adding the DragListener to the body in my subclass.   But a further requirement of BLLDropperNode is that you need to be abe to drag it while holding down the momentary button, and adding the DragListener to just the body will break that.

**Comment by pixelzoom** [comment]
To expand on the technical details and requirements here...

BLLDropperNode extends EyeDropperNode.  EyeDropperNode includes the "body" Node and the momentary button, and there's currently no way to access just the body via its API.   So BLLDropperNode adds a DragListener via `this.addInputListener`.  I could resolve the issue by making the body part of the `@protected` API, then adding the DragListener to the body in my subclass.   But a further requirement of BLLDropperNode is that you need to be able to drag it while holding down the momentary button, and adding the DragListener to just the body will break that.

This is an example of why I originally suggested a PhET-iO API that would allow you to disable _specific input listeners_, not _all input_ on a Node (which is what `inputEnabledProperty` does).

**Comment by kathy-phet** [comment]
@jonathanolson @zepumph - I know you often use "Options" patterns. Is that something that should be considered for inputEnabledProperty when it should be more specific in the listeners it impacts?

**Comment by kathy-phet** [comment]
Tagged this for PhET-iO meeting for further discusion.

**Comment by zepumph** [comment]
Wasn't there a pattern used somewhere that just created a Property called `draggingEnabledProperty`, used to control whether or not the dragListener would be enabled. I can't recall the exact issue.

If we go down that road, presumably it would be best to be able to pass in an enabledProperty to the PressListener hierarchy to achieve this.

**Comment by pixelzoom** [comment]
I can certainly add a special Property like `draggingEnabledProperty` and handle it in sim-specific code.  But that will not be using `inputEnabledProperty`, and my understand is that `inputEnabledProperty` is supposed to be the new standard for disabling input.

**Comment by pixelzoom** [comment]
Here's how this can be accomplished in PHDropperNode, with no common-code changes and by adding `draggingEnabledProperty`.   I don't like adding and removing the drag listener based on  `draggingEnabledProperty`.  And again, using something other than `inputEnabledProperty` is going to make `dropperNode` different for the instructional designer. 

```diff
Index: js/common/view/PHDropperNode.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/view/PHDropperNode.js b/js/common/view/PHDropperNode.js
--- a/js/common/view/PHDropperNode.js	(revision b72b40725c7ed74b08a3d39b0543614376ae5349)
+++ b/js/common/view/PHDropperNode.js	(date 1617729370709)
@@ -8,6 +8,7 @@
  * @author Chris Malley (PixelZoom, Inc.)
  */
 
+import BooleanProperty from '../../../../axon/js/BooleanProperty.js';
 import Property from '../../../../axon/js/Property.js';
 import merge from '../../../../phet-core/js/merge.js';
 import EyeDropperNode from '../../../../scenery-phet/js/EyeDropperNode.js';
@@ -50,13 +51,27 @@
     // dilate touch area
     this.touchArea = this.localBounds.dilatedX( 0.25 * this.width );
 
-    // move the dropper
-    this.addInputListener( new DragListener( {
+    // For disabling dragging via PhET-iO, which cannot be done in this case using inputEnabledProperty.
+    // See https://github.com/phetsims/ph-scale/issues/215
+    const draggingEnabledProperty = new BooleanProperty( true, {
+      tandem: options.tandem.createTandem( 'draggingEnabledProperty' )
+    } );
+
+    const dragListener = new DragListener( {
       positionProperty: dropper.positionProperty,
       dragBoundsProperty: new Property( dropper.dragBounds ),
       transform: modelViewTransform,
       tandem: options.tandem.createTandem( 'dragListener' )
-    } ) );
+    } );
+
+    draggingEnabledProperty.link( draggingEnabled => {
+      if ( draggingEnabled ) {
+        this.addInputListener( dragListener );
+      }
+      else {
+        this.removeInputListener( dragListener )
+      }
+    } );
   }
 }
```

**Comment by pixelzoom** [comment]
Thinking as an instructional designer... It would make a lot more sense to me if I could change something like `phScale.macroScreen.view.dropperNode.dragListener.enabledProperty` to `false`.  That makes it very clear that I am disabling dragging. But listeners have no such `enabledProperty`.

**Comment by jonathanolson** [comment]
PressListener can be given a `canStartPress` option that could check draggingEnabledProperty (instead of adding/removing the listener).

It sounds like it would be helpful to potentially give PressListener an enabledProperty, such that it wouldn't start a press/drag if that Property is false. It's very similar in use to `canStartPress`, however having both seems needed in some cases.

This seems potentially complementary to the `inputEnabled` approach. `inputEnabled` seems like a good way of saying "you can't interact with this component at all", and then if you need more fine-grained control, disabling specific listeners sounds helpful.

@pixelzoom, @zepumph thoughts?

**Comment by zepumph** [comment]
> This seems potentially complementary to the `inputEnabled` approach. `inputEnabled` seems like a good way of saying "you can't interact with this component at all", and then if you need more fine-grained control, disabling specific listeners sounds helpful.

I totally agree!

I would consider `canStartPress` a code-review-passing workaround for the moment, and we should create a new issue to add enabledProperty to PressListener. I vaguely recall waiting to add it until we needed it, so alternatively we could have this issue wait on an implemnetation of passing PressListener an enabledProperty.

I'm thinking it could be quite easy to add. I can test with `beersLawLab.concentrationScreen.view.dropperNode.dragListener.enabledProperty` un studio after adding this patch:

<details>

```diff
Index: js/listeners/PressListener.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/listeners/PressListener.js b/js/listeners/PressListener.js
--- a/js/listeners/PressListener.js	(revision e3a4962abd172ddd4e1bf0fc391645ac663e1d0f)
+++ b/js/listeners/PressListener.js	(date 1617732624022)
@@ -27,6 +27,7 @@
 import DerivedProperty from '../../../axon/js/DerivedProperty.js';
 import stepTimer from '../../../axon/js/stepTimer.js';
 import merge from '../../../phet-core/js/merge.js';
+import EnabledComponent from '../../../sun/js/EnabledComponent.js';
 import EventType from '../../../tandem/js/EventType.js';
 import PhetioObject from '../../../tandem/js/PhetioObject.js';
 import Tandem from '../../../tandem/js/Tandem.js';
@@ -42,7 +43,7 @@
 // Factor out to reduce memory footprint, see https://github.com/phetsims/tandem/issues/71
 const truePredicate = _.constant( true );
 
-class PressListener {
+class PressListener extends EnabledComponent {
   /**
    * @param {Object} [options] - See the constructor body (below) for documented options.
    */
@@ -138,6 +139,8 @@
     assert && assert( typeof options.a11yLooksPressedInterval === 'number',
       'a11yLooksPressedInterval should be a number' );
 
+    super( options );
+
     // @private {number} - Unique global ID for this listener
     this._id = globalID++;
 
@@ -350,7 +353,7 @@
    * @returns {boolean}
    */
   canPress( event ) {
-    return !this.isPressed && this._canStartPress( event, this ) &&
+    return this.enabledProperty.value && !this.isPressed && this._canStartPress( event, this ) &&
            // Only let presses be started with the correct mouse button.
            ( !( event.pointer instanceof Mouse ) || event.domEvent.button === this._mouseButton ) &&
            // We can't attach to a pointer that is already attached.
@@ -367,7 +370,7 @@
   canClick() {
     // If this listener is already involved in pressing something (or our options predicate returns false) we can't
     // press something.
-    return !this.isPressed && this._canStartPress( null, this );
+    return this.enabledProperty.value && !this.isPressed && this._canStartPress( null, this );
   }
 
   /**

```

</details>

Next, if we decide that not ALL PressListeners should get an instrumented enabledProperty, we could create an opt in like so:

<details>

```diff
Index: sun/js/EnabledComponent.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/sun/js/EnabledComponent.js b/sun/js/EnabledComponent.js
--- a/sun/js/EnabledComponent.js	(revision 2d9d1fe549192eb357ea9da9eb65f5909300a0cb)
+++ b/sun/js/EnabledComponent.js	(date 1617732772542)
@@ -27,7 +27,10 @@
   enabledPropertyOptions: null,
 
   // phet-io
-  tandem: Tandem.OPTIONAL
+  tandem: Tandem.OPTIONAL,
+
+  // When false, do not instrument the default enabledProperty created in this type.
+  phetioInstrumentEnabledProperty: true
 };
 
 class EnabledComponent {
@@ -42,7 +45,7 @@
 
     // @public
     this.enabledProperty = options.enabledProperty || new EnabledProperty( options.enabled, merge( {
-      tandem: options.tandem.createTandem( EnabledProperty.TANDEM_NAME )
+      tandem: options.phetioInstrumentEnabledProperty ? options.tandem.createTandem( EnabledProperty.TANDEM_NAME ) : Tandem.OPTIONAL
     }, options.enabledPropertyOptions ) );
 
     // @private - called by dispose
Index: beers-law-lab/js/concentration/view/BLLDropperNode.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/beers-law-lab/js/concentration/view/BLLDropperNode.js b/beers-law-lab/js/concentration/view/BLLDropperNode.js
--- a/beers-law-lab/js/concentration/view/BLLDropperNode.js	(revision 985bbbd5e8d3bc3077d0705ec6a587da12f4c057)
+++ b/beers-law-lab/js/concentration/view/BLLDropperNode.js	(date 1617732772540)
@@ -130,7 +130,8 @@
       positionProperty: dropper.positionProperty,
       dragBoundsProperty: new Property( dropper.dragBounds ),
       modelViewTransform: modelViewTransform,
-      tandem: options.tandem.createTandem( 'dragListener' )
+      tandem: options.tandem.createTandem( 'dragListener' ),
+      phetioInstrumentEnabledProperty: true
     } );
     this.addInputListener( dragListener );
 

```


</details>


@jonathanolson to review implementation in PressListener. Is this too much bloat in the general case?


@samreid and @pixelzoom to review or discuss if this behavior should be an "opt in" approach to limit the number of PressListener that have an enabledProperty.

**Comment by arouinfar** [comment]
Looks like I'm a bit late to the party, but I wanted to clarify this point:

> But a further requirement of PHDropperNode is that you need to be able to drag it while holding down the momentary button, and adding the DragListener to just the body will break that.

This is not a design requirement. I don't see why a user would need to drag the dropper by its button, especially since it's a large UI element and can be grabbed elsewhere.  There isn't a pedagogical reason why you need to be able to drag the dropper while it is dispensing the solute, either. Similarly, the laser in Bending Light cannot be dragged by its button (https://github.com/phetsims/scenery-phet/issues/660#issuecomment-772618956).

**Comment by arouinfar** [comment]
"More fun" is fine if there aren't huge technical costs, which doesn't seem to be the case here.

**Comment by pixelzoom** [comment]
> @samreid and @pixelzoom to review or discuss if this behavior should be an "opt in" approach to limit the number of PressListener that have an enabledProperty.

I think this is a question for designers.  From an implementation point of view, it would be easier to opt out, and only feature the input listeners that designers think might be desirable to disable individually.

**Comment by pixelzoom** [comment]
> "More fun" is fine if there aren't huge technical costs, which doesn't seem to be the case here.

This problem and the associated cost are not specific to `dropperNode`.  There is a much more general problem -- the inability to disable individual input listeners, and the fact that `inputEnbledProperty` is "all or nothing".  Changing the behavior of the dropper will do nothing to make that general problem go away, it will just pop up in a different context.  So I don't see a lot of value in changing the dropper behavior.

**Comment by jonathanolson** [comment]
The top patch in https://github.com/phetsims/ph-scale/issues/215#issuecomment-814336290 looks generally good, but PressListener would need to add a `super.dispose()` in its `dispose()`. With that change, it looks good to me, and I'm not worried about the bloat.

I'll rely on others to figure out the phet-io details.

**Comment by pixelzoom** [comment]
@jonathanolson said:
> This seems potentially complementary to the `inputEnabled` approach. `inputEnabled` seems like a good way of saying "you can't interact with this component at all", and then if you need more fine-grained control, disabling specific listeners sounds helpful.
> 
> @pixelzoom, @zepumph thoughts?

I like this direction, +1.

**Comment by pixelzoom** [comment]
So regarding the "phet-io details"...   

For the case of `dropperNode` in ph-scale, my understanding is that `dropperNode` branch of the Studio tree would look like this:

```
-dropperNode
   inputEnabledProperty
   visibleProperty
  -button
   *enabledProperty
    visibleProperty
   -pressListener
      enabledProperty
      pressAction
      releaseAction
 -dragListener
    enabledProperty
    dragAction
    pressAction
    releaseAction
```

To disable all interaction with `dropperNode` (dragging and button), set `dropperNode.inputEnabledProperty` to false.

To only disable dragging, set `dropperNode.dragListener.enabledProperty` to false.

To only disable interaction with the button, you have 4 choices:
* set `dropperNode.button.visibleProperty` to false (probably makes the most sense)
* set `dropperNode.button.enabledProperty` to false
* instrument `dropperNode.button.inputEnabledProperty` and set it to false
* set `dropperNode.button.pressListener.enbledProperty` to false

Do I understand this correctly?

**Comment by pixelzoom** [comment]
Other questions...

* Should sun buttons opt out of `enabledProperty`, so that we don't have `button.enabledProperty` and `button.pressListener.enabledProperty`? 
* Should we be able to pass `enabledProperty` to an input listener via options? (like Properties to Node)
* If `options.enabledProperty` is supported, can it be a DerivedProperty, or must it be settable?

**Comment by jonathanolson** [comment]
> Should we be able to pass enabledProperty to an input listener via options? (like enabledProperty et. al. to Node)

That's my understanding of what the patch does, and I would like that.

> If options.enabledProperty is supported, can it be a DerivedProperty, or must it be settable?

I would have a strong preference for supporting DerivedProperty.

**Comment by zepumph** [comment]
> > If options.enabledProperty is supported, can it be a DerivedProperty, or must it be settable?
> 
> I would have a strong preference for supporting DerivedProperty.

The current implementation (although untested) should support being a DerivedProperty. If we commit, unit tests should accompany it to make sure that behavior remains.



------------------

My **strong** preference here is for phet-io instrumenting `enabledProperty` to be an opt in for PressListener. 


>And my understand is that `inputEnabledProperty` is supposed to be the new standard for disabling input.


I totally agree. `PressListener.enabledProperty` to me is a supplemental way to control specific cases that need more control that inputEnabledProperty can allow. In previous discussions we have determined for a variety of reasons that it could be quite buggy to disable one input listener and not another (like when multiple inputs, keyboard/mouse, control the same object with different listeners). I don't love the idea of adding PressListener.enabledProperty everywhere, and then spending time figuring out all the places that we need to opt out of this. We should use `PressListener.enabledProperty` as a scalpel to solve the problems in front of us, and `inputEnabledProperty` as the hammer (which works most of the time).

**Comment by samreid** [comment]
In today's phet-io meeting, we agreed we would like the fine-grained control, so that PressListener would have `enabledProperty`.  We would use an option pattern like `Node.enabledPropertyPhetioInstrumented` so that we can decide on a case-by-case basis whether the `PressListener.enabledProperty` would be instrumented.

@zepumph, please take the lead and coordinate with @jonathanolson.  We would like this to be ready for the upcoming pH-scale simulation phet-io release, which we would like to publish by May 30.

**Comment by arouinfar** [comment]
May 30 is getting pretty close, so tagging for phet-io meeting to check in on this issue.

**Comment by arouinfar** [comment]
On Slack @pixelzoom said:

> Since the goal is to limit master changes between dev and RC tests, I highly recommend addressing both of those issues before dev testing.  https://github.com/phetsims/ph-scale/issues/215 will certainly involve significant changes to common code, and possibly to sim code, and will be challenging to patch into a release branch. https://github.com/phetsims/studio/issues/95 seems specific to Studio, and would (hopefully) be straightforward to patch into a release branch.

**Comment by pixelzoom** [comment]
See also https://github.com/phetsims/ph-scale/issues/141 (dropper: disable dragging without disabling button)

**Comment by zepumph** [comment]
My plan is to get to this 5/5, but we'll see how that goes.

**Comment by zepumph** [comment]
I went with the opt-in approach in the first `details` in https://github.com/phetsims/ph-scale/issues/215#issuecomment-814336290

It went really smoothly for almost everything. Now every `PressListener` has an `enabledProperty`, that is, by default, not instrumented, but can be opted into with `phetioEnabledPropertyInstrumented: true` (like I did for the `DragListener` in `PHDropperNode`). 


I tested this in 4 ways.

1. With the above commit to the `PHDropperNode`, I was able to go to studio, toggle the `enabledProperty` on the `dragListener`, and was not able to drag the Node, though the button still works. I believe that was the required design.
2. I temporarily set the `PressListener` default from false to `phetioEnabledPropertyInstrumented:true`, and regenerated an API file. I found a new `enabledProperty` under each `PressListener` (or sub type).
3. With this patch, every sun button doesn't work, don't commit this :).
<details>

```diff

Index: js/buttons/ButtonNode.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/buttons/ButtonNode.js b/js/buttons/ButtonNode.js
--- a/js/buttons/ButtonNode.js	(revision 1c65b742f539ae32e4cfc4ebd0410470dcb7c3d8)
+++ b/js/buttons/ButtonNode.js	(date 1620245714214)
@@ -133,6 +133,7 @@
 
     // @private {PressListener}
     this._pressListener = buttonModel.createPressListener( options.listenerOptions );
+    this._pressListener.enabledProperty.value = false;
     this.addInputListener( this._pressListener );
 
     assert && assert( buttonBackground.fill === null, 'ButtonNode controls the fill for the buttonBackground' );
```

</details>


4. With this patch, the hands in Ratio and Proportion don't work for mouse/touch. 

<details>

```diff

Index: js/common/view/RatioHalf.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/view/RatioHalf.js b/js/common/view/RatioHalf.js
--- a/js/common/view/RatioHalf.js	(revision 01e40100406e65ea9e79bb9a025e2e8637f30d31)
+++ b/js/common/view/RatioHalf.js	(date 1620245644379)
@@ -324,6 +324,7 @@
       tandem: config.tandem.createTandem( 'dragListener' )
     } );
 
+    dragListener.enabledProperty.value = false;
     // When the range changes, update the dragBounds of the drag listener
     config.enabledRatioTermsRangeProperty.link( enabledRange => {
       const newBounds = getModelBoundsFromRange( enabledRange );

```
</details>




The main outstanding question is how to handle the cursor, because hit-testing doesn't respect an `inputListener` is disabled, and there is no recalculating based on toggling `enabledProperty`. I don't know a good way to set the cursor back to default on a Node with only disabled input listeners, or even if that is desirable (presumably it would be). 

@jonathanolson please take a look at the

**Comment by jonathanolson** [comment]
First thing of note is that EnabledComponent is in sun, scenery can't depend on that at the moment (breaks scenery builds). Can we move EnabledComponent to scenery?

**Comment by jonathanolson** [comment]
> Can we move EnabledComponent to scenery?

Actually, axon, since it doesn't require anything else?

**Comment by jonathanolson** [comment]
PressListener will need a `super.dispose()` in its dispose.

**Comment by jonathanolson** [comment]
If a PressListener is disabled, it would probably be best if we interrupt it. Does that sound like a good plan?

**Comment by jonathanolson** [comment]
> 'incompatible options. Cannot specify phetioEnabledPropertyInstrumented opt out and a Tandem'

Presumably this would just be a tandem in enabledPropertyOptions, NOT the tandem for the EnabledComponent, could we clarify that documentation? (I presume we don't need to worry about specifying `enabledPropertyOptions: { tandem: Tandem.OPT_OUT }`, that would fail the assertion but I don't think it's worth a change)

**Comment by jonathanolson** [comment]
Looks good, see the above!

**Comment by zepumph** [comment]
Done above, please review commits in https://github.com/phetsims/axon/issues/354 as well, and reopen over there if you have comments about those commits.

**Comment by zepumph** [comment]
Can you also please comment on how to proceed with `cursor` setting?

**Comment by jonathanolson** [comment]
> Can you also please comment on how to proceed with cursor setting?

I feel like this is kind of a bigger issue... since the cursor is in practice tied to the input listener, I think looking into creating logic in listeners that handles cursors (and unsets that when they are disabled) is the proper way to go. Presumably we could add the ability (but NOT switch things over) at first, so in cases like this we can move the cursor logic from "putting it on the node" to "putting it on the listener, which has a reference to the node".

Commits look good.

**Comment by zepumph** [comment]
@jonathanolson and I were able to have an impromptu pairing session (aka scenery toe-dip) where we worked on a new feature of Node to use inputListeners to get its cursor value. See https://github.com/phetsims/scenery/issues/1219. Because of it, we were able to opt into this new feature for PHDropperNode in the above commit. Basically, because we no longer get the cursor from the Node, but instead from the DragListener, we can clear the cursor when that DragListener is disabled. 

From here we are ready for design review here. Please play with `phScale.macroScreen.view.dropperNode.dragListener.enabledProperty` in studio
 and let me know what you think!

**Comment by pixelzoom** [comment]
I had a Slack conversation with @jonathanolson about the new options that are being used here, for listener-specific cursors.  I'm not a fan of moving the cursor into listeners, because I think that it further complicates the listener API.    I won't reproduce the entire discussion here, since it's not specific to this ph-scale issue.  Our concluding comments were:

> Chris Malley  12:47 PM
... Just pointing out that there is now a hierarchy of listeners, affected by implicit ordering, and that now also affects the cursor.  It’s a complicated API.  I may be wrong, but I’m pretty certain that most devs still have trouble understanding `attach: false`.  And listener-specific cursors moves things further in that direction.    That said, I’ll get back to reviewing the ph-scale specific issue, and whether it meets the behavioral requirements for PhET-iO.
> 
> Jonathan Olson  12:48 PM
Agreed, I'll think on ways to simplify this.

**Comment by pixelzoom** [comment]
@zepumph said:
> From here we are ready for design review here. Please play with `phScale.macroScreen.view.dropperNode.dragListener.enabledProperty` in studio
and let me know what you think!

It's working as expected.   If I set `dropperNode.dragListener.enabledProperty` to `false`, I can no longer drag the dropper, and there is no cursor that leads me to believe that I can drag it.  The button still works, and still has the expected cursor. 

@arouinfar  If this looks OK to you, please assign back to me so that I can prepare this issue for QA.

**Comment by arouinfar** [comment]
This looks good in master, and I agree with @pixelzoom's  https://github.com/phetsims/ph-scale/issues/215#issuecomment-837149407

**Comment by pixelzoom** [comment]
Instructions for verifying:

1. Run the sim in Studio
2. Go to the Macro screen. Verify that the dropper can be dragged by its body or button, and that dragging and the button work as expected.
3. Go to element `phScale.macroScreen.view.dropperNode.dragListener.enabledProperty`
4. Set to `false` and verify that the dropper cannot be dragged, but the button still works as expected.
5. Set to `true` and verify that the dropper behaves as it did in step 2.

**Comment by KatieWoe** [comment]
Steps above work on Win 10 Chrome in 1.5.0-dev.1 Closing
